### PR TITLE
Adding GetAzureRmRecoveryServicesVaultBackupCredentials

### PR DESCRIPTION
**Pull request by pikumarmsft**

### CODE REVIEW

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Common/PSRecoveryServicesVaultExtendedInfoClient.cs [hunk 1]
```diff
@@ -126,6 +126,27 @@ public ASRVaultCreds GenerateVaultCredential(X509Certificate2 managementCert, AR
...
+        /// Upload cert to idmgmt
+        /// </summary>
+        /// <param name="managementCert"></param>
+        /// <returns></returns>
+        public UploadCertificateResponse UploadCertificate(X509Certificate2 managementCert, ARSVault vault)
+        {
+            var certificateArgs = new CertificateArgs();
+            certificateArgs.Properties = new Dictionary<string, string>();
+            certificateArgs.Properties.Add("certificate", Convert.ToBase64String(managementCert.GetRawCertData()));
+            // CertificateArgs.Properties.Add("ContractVersion", "V2012_12");
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Common/PSRecoveryServicesVaultExtendedInfoClient.cs]: remove this commented code

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs [hunk 2]
```diff
@@ -25,21 +25,25 @@ namespace Microsoft.Azure.Commands.RecoveryServices
...
         /// <summary>
         /// Expiry in hours for generated certificate.
         /// </summary>
-        private const int VaultCertificateExpiryInHoursForHRM = 120; 
+        private const int VaultCertificateExpiryInHoursForHRM = 120;
+
+        /// <summary>
+        /// Expiry in hours for generated certificate.
+        /// </summary>
+        private const int VaultCertificateExpiryInHoursForBackup = 48;
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs:46]: it is good if you can define some constant variables

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs [hunk 3]
```diff
@@ -63,10 +67,38 @@ public class GetAzureRmRecoveryServicesVaultSettingsFile : RecoveryServicesCmdle
...
+        [Parameter(Mandatory = true, Position = 2)]
         public string Path { get; set; }
 
+        /// <summary>
+        /// Gets or sets the path where the credential file is to be generated
+        /// </summary>
+        /// <summary>
+        /// Gets or sets vault Object.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByDefault, Mandatory = true, ValueFromPipeline = true)]
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs]: do you want to take this switch param as well from pipeline?

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs [hunk 4]
```diff
@@ -63,10 +67,38 @@ public class GetAzureRmRecoveryServicesVaultSettingsFile : RecoveryServicesCmdle
...
+        private bool asr;
+
+        /// <summary>
+        /// Gets or sets the path where the credential file is to be generated
+        /// </summary>
+        /// <summary>
+        /// Gets or sets vault Object.
+        /// </summary>
+        [Parameter(ParameterSetName = "ForBackup", Mandatory = true)]
+        public SwitchParameter Backup
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs:102]: pls test these param sets, position numbers manually from powershell.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesVaultBackupCredentials.cs [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+namespace Microsoft.Azure.Commands.RecoveryServices
+{
+    public partial class GetAzureRmRecoveryServicesVaultSettingsFile
+    {
+        public void GetAzureRMRecoveryServicesVaultBackupCredentials(string targetLocation, int vaultCertificateExpiryInHoursForBackup)
+        {
+            if (!Directory.Exists(targetLocation))
+            {
+                throw new ArgumentException("The target location provided is not a directory. Please provide a directory.");
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesVaultBackupCredentials.cs]: do you want pick these messages from resx file?

**Comment by sriramvu** [comment]
- make help file changes.
- please do basic validation with site recovery cmdlets.
- please inform / share change in the cmdlet as it works only with one of the switch parameters. One of the Azure articles which talks about the cmdlet: https://azure.microsoft.com/en-in/documentation/articles/site-recovery-deploy-with-powershell-resource-manager/

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Commands.RecoveryServices.csproj [hunk 6]
```diff
@@ -129,6 +129,9 @@
     </Compile>
     <Compile Include="Utilities\CertUtils.cs" />
     <Compile Include="Utilities\Utilities.cs" />
+    <Compile Include="Vault\GetAzureRmRecoveryServicesVaultBackupCredentials.cs">
```
Comment (sriramvu) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Commands.RecoveryServices.csproj]: shall we have some helper or so, as we maintain the cmdlet files here.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Common/PSRecoveryServicesVaultExtendedInfoClient.cs [hunk 7]
```diff
@@ -126,6 +126,27 @@ public ASRVaultCreds GenerateVaultCredential(X509Certificate2 managementCert, AR
...
+
+            var response = this.recoveryServicesClient.VaultExtendedInfo.UploadCertificateAsync(
+                vault.ResouceGroupName,
+                vault.Name,
+                certificateArgs, managementCert.FriendlyName,
+                this.GetRequestHeaders()); ;
```
Comment (sriramvu) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Common/PSRecoveryServicesVaultExtendedInfoClient.cs]: nit: extra ;

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesVaultBackupCredentials.cs [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+            }
+        }
+
+        /// <summary>
+        /// Generates vault creds file content for backup Vault
+        /// </summary>
+        /// <param name="cert">management certificate</param>
+        /// <param name="subscriptionId">subscription Id</param>
+        /// <param name="resourceType">resource type</param>
+        /// <param name="displayName">display name</param>
```
Comment (sriramvu) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesVaultBackupCredentials.cs]: nit: please keep the required params. (cross check in other method summaries too)

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Commands.RecoveryServices.csproj [hunk 9]
```diff
@@ -165,4 +168,5 @@
     </Content>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <Import Project="$(SolutionDir)\.nuget\NuGet.targets" Condition="Exists('$(SolutionDir)\.nuget\NuGet.targets')" />
```
Comment (sriramvu) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Commands.RecoveryServices.csproj]: Are we adding any specific NuGet targets?

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Models/PSContracts.cs [hunk 10]
```diff
@@ -358,10 +358,10 @@ public VaultCreds()
         /// <param name="resourceName">resource name</param>
         /// <param name="managementCert">management cert</param>
         /// <param name="acsNamespace">authenticating service namespace</param>
```
Comment (sriramvu) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Models/PSContracts.cs:360]: add summary for new resource type.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Models/PSContracts.cs [hunk 11]
```diff
@@ -505,6 +505,55 @@ public ASRVaultCreds()
     }
 
     /// <summary>
+    /// Class to define backup vault credentials
+    /// </summary>
+    [SuppressMessage("Microsoft.StyleCop.CSharp.MaintainabilityRules", "SA1402:FileMayOnlyContainASingleClass", Justification = "Grouping classes based on entity")]
```
Comment (sriramvu) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Models/PSContracts.cs]: have we enabled style cop?

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs [hunk 12]
```diff
@@ -25,21 +25,25 @@ namespace Microsoft.Azure.Commands.RecoveryServices
     /// Retrieves Azure Recovery Services Vault Settings File.
     /// </summary>
     [Cmdlet(VerbsCommon.Get, "AzureRmRecoveryServicesVaultSettingsFile")]
-    [OutputType(typeof(VaultSettingsFilePath))]
-    public class GetAzureRmRecoveryServicesVaultSettingsFile : RecoveryServicesCmdletBase
+    [OutputType(typeof(VaultSettingsFilePath), typeof(string))]
```
Comment (sriramvu) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs]: let's continue to return the VaultSettingsFilePath object.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs [hunk 13]
```diff
@@ -63,10 +67,38 @@ public class GetAzureRmRecoveryServicesVaultSettingsFile : RecoveryServicesCmdle
         /// <summary>
         /// Gets or sets vault Object.
         /// </summary>
-        [Parameter(ParameterSetName = ASRParameterSets.ByDefault)]
-        [Parameter(ParameterSetName = ASRParameterSets.ForSite)]
+        [Parameter(Mandatory = true, Position = 2)]
```
Comment (sriramvu) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRMRecoveryServicesVaultSettingsFile.cs]: let this be optional.

**Pull request merged**